.github/workflows/bulk-dependency-updates.yaml

```diff
@@ -0,0 +1,119 @@
+        description: 'The default GITHUB_TOKEN for your workflow.'
+      VAULT_ECO_GITHUB_TOKEN:
+        required: true
+        description: 'Should be different than the default GITHUB_TOKEN so that this workflow will trigger checks on the resulting PR.'
+
+jobs:
+  upgrade:
+    name: Upgrade & Open Pull Request
+    runs-on: ubuntu-latest
+    env:
+      # This branch will receive updates each time the workflow runs
+      # It doesn't matter if it's deleted when merged, it'll be re-created
+      BRANCH_NAME: auto-dependency-upgrades
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version-file: .go-version
+          cache: true
+          # Use a separate token to automatically execute checks on the resulting PR
+          # https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow
+          token: ${{ secrets.VAULT_ECO_GITHUB_TOKEN }}
+
+      - name: Upgrade Go dependencies
+        run: |
+          go list -u -m -json all | jq -r 'select(.Indirect != true and .Update != null) | .Path+"@"+.Update.Version' | xargs -L1 go get
+          go mod tidy
+
+      - name: Detect changes
+        id: changes
+        run:
+          # This output boolean tells us if the dependencies have actually changed
+          echo "count=$(git status --porcelain=v1 2>/dev/null | wc -l)" >> "$GITHUB_OUTPUT"
+
+      - name: Commit & push changes
+        # Only push if changes exist
+        if: steps.changes.outputs.count > 0
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add .
+          git commit -m "Automated dependency upgrades"
+          git push -f origin ${{ github.ref_name }}:"$BRANCH_NAME"
+
+      - name: Open pull request if needed
+        if: steps.changes.outputs.count > 0
+        env:
+          GH_TOKEN: ${{ secrets.REPO_GITHUB_TOKEN }}
+          REVIEWER_TEAM: ${{ inputs.reviewer-team }}
+          REPO: ${{ inputs.repository }}
+          RUN_ID: ${{ inputs.run-id }}
+        # Only open a PR if the branch is not attached to an existing one
+        run: |
+          PR=$(gh pr list --head "$BRANCH_NAME" --json number -q '.[0].number')
+
+          # currently unable to set team as reviewer in GHA
+          # see https://github.com/cli/cli/issues/6395
+          reviewers=""
+          if [[ "$REVIEWER_TEAM" == "hashicorp/vault-ecosystem-applications" ]]; then
+            reviewers="austingebauer,fairclothjm,vinay-gopalan,maxcoulombe,robmonte,Zlaticanin,kpcraig,raymonstah"
+          elif [[ "$REVIEWER_TEAM" == "hashicorp/vault-ecosystem-foundations" ]]; then
+            reviewers="swenson,benashz,tvoran,tomhjp,kschoche,thyton"
+          else
+            echo "Unknown input for reviewer-team"
+            exit 1
```

Comment:
We don't have to fail here. I guess if it causes problems we can change it later